soh/src/code/z_message_PAL.c

```diff
@@ -1611,6 +1634,9 @@ void Message_Decode(PlayState* play) {
             decodedBufPos--;
         } else if (temp_s2 == MESSAGE_ITEM_ICON) {
             msgCtx->msgBufDecoded[++decodedBufPos] = font->msgBuf[msgCtx->msgBufPos + 1];
+            if (msgCtx->msgBufDecoded[decodedBufPos] == ITEM_INVALID_4) {
+                msgCtx->msgBufDecoded[++decodedBufPos] = font->msgBuf[msgCtx->msgBufPos + 2];
+            }
```

Comment:
this makes sure we have the character we use to determine the custom item index in the decoded buffer